lib/puppet/ssl/host.rb

```diff
@@ -419,6 +430,138 @@ def load_crls(path)
     end
   end
 
+  # Ensures that the CA certificate is available for either generating or
+  # validating the host's cert.
+  # It will first check if the cert is present in memory (used for testing),
+  # then check on disk, and finally try to download it.
+  # @return true if the CA certificate was found, false otherwise
+  def ensure_ca_certificate
+    file_path = certificate_location(CA_NAME)
+    if check_for_certificate_in_memory(CA_NAME)
+      true
+    elsif Puppet::FileSystem.exist?(file_path)
+      load_certificate_bundle(Puppet::FileSystem.read(file_path))
+      true
+    else
+      bundle = download_ca_certificate_bundle
+      if bundle
+        save_certificate_bundle(bundle)
+        true
+      else
+        false
+      end
+    end
+  end
+
+  # Creates an arry of SSL Certificate objects from a PEM-encoding string
+  # of one or more certs.
+  # @param [String] bundle_string PEM-encoded string of certs
+  # @return [[OpenSSL::X509::Certificate]] the certs loaded from the input string
+  def load_certificate_bundle(bundle_string)
```

Comment:
One thing I noticed about this method of getting an array of certs—which is essentially duped [here](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/validator/default_validator.rb#L127-L137)—is that it will remove any comments that are served. Maybe we don't care, but I did get rid of some artifacts that we had for testing [here](https://github.com/puppetlabs/puppet/pull/6806/files#diff-36ba91bafadfe988c92ed0a8438779b6L99), so it might something to make sure we don't care about?